modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/impl/DDMFormInstanceRecordLocalServiceImpl.java

```diff
@@ -241,6 +245,52 @@ public DDMFormInstanceRecord deleteFormInstanceRecord(
 			DDMFormInstanceRecord ddmFormInstanceRecord)
 		throws PortalException {
 
+		return deleteFormInstanceRecord(ddmFormInstanceRecord, true);
+	}
+
+	@Indexable(type = IndexableType.DELETE)
+	@Override
+	@SystemEvent(
+		action = SystemEventConstants.ACTION_SKIP,
+		type = SystemEventConstants.TYPE_DELETE
+	)
+	public DDMFormInstanceRecord deleteFormInstanceRecord(
+			DDMFormInstanceRecord ddmFormInstanceRecord, boolean updateReport)
+		throws PortalException {
+
+		try {
+			if ((ddmFormInstanceRecord.getStatus() ==
+					WorkflowConstants.STATUS_APPROVED) &&
+				updateReport) {
+
+				DDMFormInstanceReport ddmFormInstanceReport =
+					_ddmFormInstanceReportLocalService.
+						getFormInstanceReportByFormInstanceId(
+							ddmFormInstanceRecord.getFormInstanceId());
+
+				DDMFormInstanceRecordVersion ddmFormInstanceRecordVersion =
+					ddmFormInstanceRecord.getLatestFormInstanceRecordVersion();
+
+				_ddmFormInstanceReportLocalService.
+					processFormInstanceReportEvent(
+						ddmFormInstanceReport.getFormInstanceReportId(),
+						ddmFormInstanceRecordVersion.
+							getFormInstanceRecordVersionId(),
+						DDMFormInstanceReportConstants.
+							EVENT_DELETE_RECORD_VERSION);
+			}
+		}
+		catch (Exception exception) {
+			if (_log.isWarnEnabled()) {
+				_log.warn(
+					StringBundler.concat(
+						"Unable to update dynamic data mapping form instance ",
+						"report for dynamic data mapping form instance record ",
+						ddmFormInstanceRecord.getFormInstanceRecordId()),
+					exception);
+			}
+		}
```

Comment:
I would:

1. Remove this TRY/CATCH because the method already throws an exception.
2. Can we move this logic to the end of the method instead of the beginning? Can you test the scenario of creating a form, adding entries (a report will be created automatically; see documentation), deleting one entry, and seeing if the report will be broken because we are updating only after the deletion?